tests/PHPCurlClass/PHPCurlClassTest.php

```diff
@@ -1083,1193 +1075,91 @@ public function testJSONDecoder()
 
     public function testJSONContentTypeDetection()
     {
-        $json_content_types = array(
-            'application/alto-costmap+json',
-            'application/alto-costmapfilter+json',
-            'application/alto-directory+json',
-            'application/alto-endpointprop+json',
-            'application/alto-endpointpropparams+json',
-            'application/alto-endpointcost+json',
-            'application/alto-endpointcostparams+json',
-            'application/alto-error+json',
-            'application/alto-networkmapfilter+json',
-            'application/alto-networkmap+json',
-            'application/calendar+json',
-            'application/coap-group+json',
-            'application/jose+json',
-            'application/jrd+json',
-            'application/json',
-            'application/json-patch+json',
-            'application/json-seq',
-            'application/jwk+json',
-            'application/jwk-set+json',
-            'application/ld+json',
-            'application/merge-patch+json',
-            'application/rdap+json',
-            'application/reputon+json',
-            'application/vcard+json',
-            'application/vnd.apache.thrift.json',
-            'application/vnd.api+json',
-            'application/vnd.bekitzur-stech+json',
-            'application/vnd.collection.doc+json',
-            'application/vnd.collection+json',
-            'application/vnd.collection.next+json',
-            'application/vnd.document+json',
-            'application/vnd.geo+json',
-            'application/vnd.hal+json',
-            'application/vnd.heroku+json',
-            'application/vnd.ims.lis.v2.result+json',
-            'application/vnd.ims.lti.v2.toolconsumerprofile+json',
-            'application/vnd.ims.lti.v2.toolproxy.id+json',
-            'application/vnd.ims.lti.v2.toolproxy+json',
-            'application/vnd.ims.lti.v2.toolsettings+json',
-            'application/vnd.ims.lti.v2.toolsettings.simple+json',
-            'application/vnd.mason+json',
-            'application/vnd.micro+json',
-            'application/vnd.miele+json',
-            'application/vnd.oftn.l10n+json',
-            'application/vnd.oracle.resource+json',
-            'application/vnd.siren+json',
-            'application/vnd.xacml+json',
-
-            // Non-standard
-            'application/x-json',
-            'text/json',
-            'text/x-json',
-        );
+        $test = new Test();
+        $not_json_content_types = array();
+        $json_content_types = $test->getContentTypes('json',
+            $not_json_content_types);
 
-        $class = new ReflectionClass('\Curl\Curl');
-        $property = $class->getProperty('jsonPattern');
```

Comment:
This is the only "breaking" change.  The JSON and XML patterns are now stored differently, as evidenced here.  This increases accessibility, but it may cause if an extending class uses this property.  As this was a protected property already, this will have no effect for 99.9% of use-cases.